src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotHardCodeEncryptionKeyTests.cs

```diff
@@ -312,6 +312,72 @@ public void TestMethod(byte[] someOtherBytesForIV)
             GetCSharpResultAt(12, 9, 10, 26, "ICryptoTransform SymmetricAlgorithm.CreateEncryptor(byte[] rgbKey, byte[] rgbIV)", "void TestClass.TestMethod(byte[] someOtherBytesForIV)", "byte[,]", "void TestClass.TestMethod(byte[] someOtherBytesForIV)"));
         }
 
+        [Fact]
+        public void Test_HardcodedInJaggedArrayInitializer_CreateEncryptor_Diagnostic()
+        {
+            VerifyCSharpWithDependencies(@"
+using System;
+using System.Linq;
+using System.Security.Cryptography;
+
+class TestClass
+{
+    public void TestMethod(byte[] someOtherBytesForIV, byte unknownByte)
+    {
+        byte[][] rgbKey = new byte[3][]
+        {
+            new byte[] { 1, 2 },
+            new byte[] { 3, 4, 5 },
+            new byte[] { unknownByte }
+        };
+        SymmetricAlgorithm rijn = SymmetricAlgorithm.Create();
+        rijn.CreateEncryptor(rgbKey.Cast<byte>().ToArray(), someOtherBytesForIV);
+    }
+}",
+            GetCSharpResultAt(17, 9, 13, 13, "ICryptoTransform SymmetricAlgorithm.CreateEncryptor(byte[] rgbKey, byte[] rgbIV)", "void TestClass.TestMethod(byte[] someOtherBytesForIV, byte unknownByte)", "byte[]", "void TestClass.TestMethod(byte[] someOtherBytesForIV, byte unknownByte)"),
+            GetCSharpResultAt(17, 9, 12, 13, "ICryptoTransform SymmetricAlgorithm.CreateEncryptor(byte[] rgbKey, byte[] rgbIV)", "void TestClass.TestMethod(byte[] someOtherBytesForIV, byte unknownByte)", "byte[]", "void TestClass.TestMethod(byte[] someOtherBytesForIV, byte unknownByte)"));
+        }
+
+        [Fact]
+        public void Test_ElementTypeIsTypeParameter_NoDiagnostic()
+        {
+            VerifyCSharpWithDependencies(@"
+using System;
+
+class TestClass<T1> where T1 : struct
+{
+    public void MethodWithArrayParameter<T2>(params T2[] arr) where T2 : struct
+    {
+    }
+
+    public void TestMethod(T1 t)
+    {
+        MethodWithArrayParameter(t);
+    }
+}");
+        }
+
```

Comment:
Not that I'm super worried about someone putting a hardcoded key in a params byte[] array, but would we treat the params array as a tainted source if someone called `MethodWithArrayParameters<byte>(1, 2, 3)`?  If it's easy to handle, I think we should since that would count as a constant byte array.  If it's not easy to handle, we should at least put a comment somewhere that SourceInfo.TaintConstantArray doesn't handle params arrays.

Reply:
This case can be handled. At the point of the invocation of MethodWithArrayParameters<byte>(1, 2, 3), it will trigger the callback of OperationKind.ArrayInitializer. A test case will be added.